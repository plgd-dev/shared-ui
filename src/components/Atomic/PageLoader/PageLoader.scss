@import '../../../common/styles/colors';
@import '../../../common/styles/sizes';



#page-loader {
  position: fixed;
  top: $status_bar_height;
  right: 0;
  width: calc(100% - #{$left_panel_expanded});
  z-index: 9;

  // Loader displayed when waiting for auth status
  &.auth-loader {
    top: 0;
    width: 100%;
  }

  .bar {
    width: 100%;
    height: $page_loader_height;
    background-color: $secondary_20;
    overflow: hidden;
    position: relative;

    .progress {
      width: 0;
      height: $page_loader_height;
      background-color: $primary_100;
    }
  }

  .bar {
    .progress {
      width: 100%;
      position: relative;
      background-color: transparent !important;

      .subline {
        position: absolute;
        background-color: $primary_100;
        height: $page_loader_height;

        &.inc {
          animation: increase 2s infinite;
        }
        &.dec {
          animation: decrease 2s 0.5s infinite;
        }
      }
    }
  }
}

// Collapsed menu state
#app.collapsed {
  #page-loader {
    width: calc(100% - #{$left_panel_collapsed});
  }
}
