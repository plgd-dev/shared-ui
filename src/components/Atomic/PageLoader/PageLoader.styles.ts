import { css, keyframes } from '@emotion/react'
import { ThemeType, get } from '../_theme'

const increase = keyframes`
  from {
    left: -5%;
    width: 5%;
  }
  to {
    left: 130%;
    width: 100%;
  }
`

const decrease = keyframes`
  from {
    left: -80%;
    width: 80%;
  }
  to {
    left: 110%;
    width: 10%;
  }
`

export const pageLoader = css`
    position: fixed;
    top: 85px;
    right: 0;
    width: 100%;
    z-index: 9;
`

export const bar = css`
    width: 100%;
    height: 5px;
    overflow: hidden;
    position: relative;
`

export const progress = css`
    width: 100%;
    height: 5px;
    background-color: transparent;
`

export const subline = (theme: ThemeType) => css`
    position: absolute;
    background-color: ${get(theme, `PageLoader.bar.background`)};
    height: 5px;
`

export const inc = css`
    animation: ${increase} 2s infinite;
`

export const dec = css`
    animation: ${decrease} 2s 0.5s infinite;
`
