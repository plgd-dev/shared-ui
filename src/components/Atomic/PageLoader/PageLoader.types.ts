export type Props = {
    loading: boolean
    className?: string
    collapsed?: boolean
    noOffset?: boolean
}
