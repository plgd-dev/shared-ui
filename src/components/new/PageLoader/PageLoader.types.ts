export type Props = {
    loading: boolean
    className?: string
}
