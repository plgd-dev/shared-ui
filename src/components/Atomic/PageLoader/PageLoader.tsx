import { FC, memo } from 'react'

import { Props } from './PageLoader.types'
import * as styles from './PageLoader.styles'

const PageLoader: FC<Props> = memo(({ loading, className }) =>
    loading ? (
        <div aria-busy='true' className={className} css={styles.pageLoader} role='alert'>
            <div css={styles.bar}>
                <div css={styles.progress}>
                    <div css={[styles.subline, styles.inc]} />
                    <div css={[styles.subline, styles.dec]} />
                </div>
            </div>
        </div>
    ) : null
)

PageLoader.displayName = 'PageLoader'

export default PageLoader
