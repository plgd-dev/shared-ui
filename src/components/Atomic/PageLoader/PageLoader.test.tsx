import React from 'react'
import { render } from '../../../../test/jest-wrapper'
import PageLoader from './PageLoader'

describe('<PageLoader>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <PageLoader loading={true} />
                <PageLoader loading={false} />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
