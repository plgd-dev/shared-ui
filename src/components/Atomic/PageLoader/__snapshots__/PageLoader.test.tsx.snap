// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoader> render correctly - snapshot 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -5%;
    width: 5%;
  }

  to {
    left: 130%;
    width: 100%;
  }
}

@keyframes animation-1 {
  from {
    left: -80%;
    width: 80%;
  }

  to {
    left: 110%;
    width: 10%;
  }
}

.emotion-0 {
  position: fixed;
  top: 85px;
  left: 280px;
  width: 100%;
  z-index: 9;
}

.emotion-1 {
  width: 100%;
  height: 5px;
  overflow: hidden;
  position: relative;
}

.emotion-2 {
  width: 100%;
  height: 5px;
  background-color: transparent;
}

.emotion-3 {
  position: absolute;
  background-color: #2261AE;
  height: 5px;
  -webkit-animation: animation-0 2s infinite;
  animation: animation-0 2s infinite;
}

.emotion-4 {
  position: absolute;
  background-color: #2261AE;
  height: 5px;
  -webkit-animation: animation-1 2s 0.5s infinite;
  animation: animation-1 2s 0.5s infinite;
}

<div>
    <div
      aria-busy="true"
      class="emotion-0"
      role="alert"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          />
          <div
            class="emotion-4"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
